crates/primitives/convert/src/contract.rs

```diff
@@ -190,14 +359,14 @@ fn to_legacy_entry_points_by_type(
     Ok(LegacyEntryPointsByType { constructor, external, l1_handler })
 }
 
-/// Returns a [HashMap<EntryPointType, Vec<EntryPoint>>] from a
+/// Returns a [IndexMap<EntryPointType, Vec<EntryPoint>>] from a
 /// [LegacyEntryPointsByType]
-fn from_legacy_entry_points_by_type(entries: &LegacyEntryPointsByType) -> HashMap<EntryPointType, Vec<EntryPoint>> {
+fn from_legacy_entry_points_by_type(entries: &LegacyEntryPointsByType) -> IndexMap<EntryPointType, Vec<EntryPoint>> {
     core::iter::empty()
         .chain(entries.constructor.iter().map(|entry| (EntryPointType::Constructor, entry)))
         .chain(entries.external.iter().map(|entry| (EntryPointType::External, entry)))
         .chain(entries.l1_handler.iter().map(|entry| (EntryPointType::L1Handler, entry)))
-        .fold(HashMap::new(), |mut map, (entry_type, entry)| {
+        .fold(IndexMap::new(), |mut map, (entry_type, entry)| {
```

Comment:
Note that `IndexMap` only retains insertion order as long as we do not perform any removals. Do we mutate entry points in a way that results in deletion during execution for example?

Reply:
Interesting suggestion here, didnt know for the order changes when a remove is performed. @antiyro Any idea about this point ? do we execute any remove into LegacyEntryPointsByType ? I will say we do not due to its nature but im not sure about it